data-prepper-plugins/rss-source/src/main/java/org/opensearch/dataprepper/plugins/source/rss/RssReaderTask.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rss;
+
+import com.apptasticsoftware.rssreader.Item;
+import com.apptasticsoftware.rssreader.RssReader;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.document.Document;
+import org.opensearch.dataprepper.model.document.JacksonDocument;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class RssReaderTask implements Runnable {
```

Comment:
This class should have a unit test. And it should be relatively easy to add it. You can mock the `RssReader`, `RSSSourceConfig`, and `Buffer`. 

Some test cases that I see off the bat:

* When items are empty, the buffer is not written to (this would fail as the code currently is)
* When one item is present, it is written to the buffer
* When multiple times are present, they are all written to the buffer
* Throws exception and does not write to the buffer if the reader throws an exception
* Failing to convert an item to the Event model does not write to the buffer, but the other items from that same read do write to the buffer

Reply:
Considering that I decided to pass only the URL to the `RssReaderTask` constructor and not the entire `RSSSourceConfig`, mocking the config would not be needed anymore. Do we still want explicit testcases for this class?

Reply:
Yes, test cases are still needed, whether you mock the config or not. You would still have to mock RssReader and Buffer